### PR TITLE
Refactor Ansible new & add api for forward compat

### PR DESCRIPTION
**Pull request by asmacdo**
New code structure will simplify the kubebuilder integration and will
work with the new CLI commands `init` and `create api`

### CODE REVIEW

File: internal/scaffold/ansible/init.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package ansible
```
Comment (camilamacedo86) [inline internal/scaffold/ansible/init.go:1]: New files missing license. See: https://travis-ci.org/github/operator-framework/operator-sdk/jobs/701003849#L948-L956

File: internal/scaffold/ansible/api.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package ansible
```
Comment (camilamacedo86) [inline internal/scaffold/ansible/api.go:1]: New files missing license. See: https://travis-ci.org/github/operator-framework/operator-sdk/jobs/701003849#L948-L956

File: cmd/operator-sdk/new/cmd.go [hunk 3]
```diff
@@ -129,7 +128,25 @@ func newFunc(cmd *cobra.Command, args []string) error {
...
+			log.Fatal(err)
+		}
+		cfg := input.Config{
+			AbsProjectPath: filepath.Join(projutil.MustGetwd()),
+			ProjectName:    projectName,
+		}
+
+		if err != nil {
+			return fmt.Errorf("invalid apiVersion and kind: %v", err)
+		}
```
Comment (camilamacedo86) [inline cmd/operator-sdk/new/cmd.go:147]: We do not need to do this check here. It still made in the https://github.com/operator-framework/operator-sdk/blob/1c1386f02ea76918f10ec5d3632b0c195e91bd01/internal/flags/apiflags/flags.go#L91

When we add the helm plugin the validation also will be made there and then we can remove the APIFlags from the project. See: https://github.com/operator-framework/operator-sdk/blob/master/internal/plugins/helm/flags.go#L76

Comment (asmacdo) [reply cmd/operator-sdk/new/cmd.go:147]: Nice catch.

**Comment by asmacdo** [comment]
After discussing with the team, we have decided to leave the current `new` and `add api` as is, and so I am closing this PR.

**Pull request closed without merging**